test-application.js

```diff
@@ -7,6 +7,35 @@ async function testApplicationAPI() {
   console.log("🧪 Testing Job Application API\n");
 
   try {
+    // 0. Login to get authentication token
+    console.log("0. Logging in to get authentication token...");
+    const loginResponse = await fetch(`${API_URL}/api/auth/login`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        email: "user@example.com",
+        password: "password123",
+      }),
+    });
+
+    const loginData = await loginResponse.json();
+
+    if (!loginData.token) {
+      console.log("   ✗ Failed to login:", loginData.error);
+      return;
+    }
+
+    const authToken = loginData.token;
+    const authHeaders = {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${authToken}`,
+    };
+
+    console.log(`   ✓ Logged in successfully as ${loginData.user.email}`);
+    console.log(`   User ID: ${loginData.user.id}\n`);
```

Comment:
I'd keep this on the UI for logged in successfully, you'll know from the HTTP Response if it was successful. Best to minimise logs when happy path is being followed unless for a business reason. 

Would you feel comfortable with your email address living in a logfile in the event a data breach happened? :D